vizro-core/src/vizro/managers/_data_manager.py

```diff
@@ -69,19 +105,16 @@ def _add_component(self, component_id: ComponentID, dataset_name: DatasetName):
         self.__component_to_original[component_id] = dataset_name
 
     def _load_lazy_data(self, dataset_name: DatasetName) -> pd.DataFrame:
-        self._cache_dataset_arguments = {"iris": {"timeout": 50}, "other_dataset": {"timeout": 100}}
+        logger.debug("reloading lazy data: %s", dataset_name)
 
-        @self._cache.memoize(**self._cache_dataset_arguments.get(dataset_name, {}))
+        @self._cache.memoize(**self.__lazy_data[dataset_name]._cache_arguments)
         # timeout (including 0 -> never expires), unless -> always executes function (like
         # timeout=0.000001) and doesn't update cache.
         # timeout and unless need to depend on dataset_name
-        def inner():
-            self.__lazy_data[dataset_name]()
-
-        """Returns the original data for `dataset_name`."""
-        logger.debug("reloading lazy data: %s", dataset_name)
-        time.sleep(2.0)
-        return inner()
+        def inner(dataset):
```

Comment:
Why is this argument needed? I guess there was some error without it but not sure what? 🤔

Reply:
Because `@self._cache.memoize` is decorating inner. `dataset` needs to be there so memoize knows to cache the function + argument combination

Reply:
Got it, thank you! 👍